### ISSUE TITLE
Self-training code

### ISSUE THREAD

**Issue by PCL0725** [valeoai/SALUDA#6]
Hi, thank you for your excellent work in the paper.I really want to repeat the work you did in your paper, but I don't quite understand your modification of the cosmix code in the second step of training, that is, self-training. I don't know how you specifically modified it, can you directly provide the modified cosmix code, and I would like to say thank you in advance and wish you a smooth work.

**Comment by BjoernMichele** [comment]
Dear @PCL0725, 
Thank you very much for your interest in our work. Unfortunately, I currently do not have a modified cosmix code by hand any more. Given that cosmix is not our codebase, we have provided in https://github.com/valeoai/SALUDA/tree/main/self_training a step-by-step explanation to facilitate the adaptation of their code base. I hope this will help you, if there are any quesitons do not hesitate do ask.

**Comment by BjoernMichele** [comment]
Feel free to open the issue again if there are further questions.

**Comment by Cupcee** [comment]
I would be interested to know the hyperparameter choices you've employed for the self-training step. In particular:

1) What choice for Tau (momentum) in EMA weight update did you use? CosMix used 0.9 and 0.999 depending on dataset.
2) How often did you update the EMA weights in your experiments: every batch, or once every N batches?
3) Did you threshold teacher model's pseudo-labels with some threshold, like the CosMix authors did?
4) How long (how many epochs) did you train the student?

Thank you.
